state-chain/chains/src/sol/api.rs

```diff
@@ -415,6 +411,8 @@ impl<Environment: SolanaEnvironment> SolanaApi<Environment> {
 		let compute_limit =
 			SolTrackedData::calculate_ccm_compute_limit(gas_budget, transfer_param.asset);
 
+		let durable_nonce = Environment::nonce_account()?;
```

Comment:
Maybe I forgot to mention but I changed this deliberately. Implemented like this, if the nonce lookup fails, we will no longer have access to the ALT data (see PRO-2238)/

We have two options: 
- Get the nonce first, but put it back if the ALT lookup fails.
- Get the ALT values first, but put them back if the nonce lookup fails. 
- (Actually a third option, described in the linear issue: make tx building transactional)

Since we already have the code for `recover_durable_nonce`, I applied the first option.